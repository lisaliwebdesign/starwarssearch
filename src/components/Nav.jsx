import React from "react";
import PropTypes from "prop-types";
import {
  NavBar, // The wrapper for the entire nav (nav element)
  HomeLink, // A link (a tag) which will take the user home
  HomeImage, // Image of a house
  CharName // The name of the Character
} from "../elements/nav";
import homeImage from "../images/home.png";

export default class Nav extends React.Component {
  static propTypes = {
    name: PropTypes.string.isRequired
  };

  render() {
    return (
      <NavBar>
        <HomeLink to="/">
          <HomeImage src={homeImage} />
        </HomeLink>
        <CharName>{this.props.name}</CharName>
      </NavBar>
    );
  }
}
