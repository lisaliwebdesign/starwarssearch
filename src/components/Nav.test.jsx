import Nav from "./Nav";

it("<components><Nav> - renders component", () => {
  const wrapper = shallow(<Nav name="test" />);
  expect(wrapper).toMatchSnapshot();
});
