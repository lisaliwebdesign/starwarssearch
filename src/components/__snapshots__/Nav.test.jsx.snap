// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<components><Nav> - renders component 1`] = `
<styled.nav>
  <Styled(Link)
    to="/"
  >
    <styled.img
      src="home.png"
    />
  </Styled(Link)>
  <styled.h1>
    test
  </styled.h1>
</styled.nav>
`;
